features/steps/thens/relations.py

```diff
@@ -133,6 +133,44 @@ def step_impl(context, related, relating, other_entity, condition):
     misc.handle_errors(context, errors)
 
 
+@then('The IfcPropertySet must be assigned according to the property set definitions table {table}')
+def step_impl(context, table):
+    if getattr(context, 'applicable', True):
+        errors = []
+        tbl_path = system.get_abs_path(f"resources/property_set_definitions/{table}")
+        tbl = system.get_csv(tbl_path, return_type='dict')
+
+        property_set_definitons = {d['Name']: d['ApplicableClasses'] for d in tbl}
+        property_set_definitons = {key: value.split(';') for key, value in property_set_definitons.items()}
+
+        for inst in context.instances:
+            if isinstance(inst, (tuple, list)):
+                inst = inst[0]
+
+            name = getattr(inst, 'Name', 'Attribute not found')
+
+            if name not in property_set_definitons.keys():
+                errors.append(err.InvalidValueError(False, inst, 'Name', name))  # A custom Pset_ prefixed attribute, e.g. Pset_Mywall
+                continue
+
+            try:
+                relation = inst.PropertyDefinitionOf[0]  # IFC2x3
+            except AttributeError:
+                relation = inst.DefinesOccurrence[0]  # IFC4x3
+            except IndexError:  # IfcPropertySet not assigned, rule not further checked
+                continue
```

Comment:
I'm thinking how to standardize it. This way we could probably use it with other attribute lookups as well?
```
def get_relation(instance, attrs : list):
    relations = (
        getattr(instance, attr, [None])[0] 
        for attr in attrs
    )
    return next((rel for rel in relations if rel is not None), None) # always len == 1

relation = get_relation(inst, ['PropertyDefinitionOf', 'DefinesOccurrence'])
if relation is None:
    continue
```

Reply:
That's a very nice idea. Not sure how much we'll use it, but let me add here and see how it goes :)